File: Frontend/lib/core/models/meal.dart
class Meal {
  final int id;
  final String name;
  final String mealType;
  final int calories;
  final double? protein;
  final double? carbs;
  final double? fat;
  final DateTime mealDate;
  final String? notes;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Meal({
    required this.id,
    required this.name,
    required this.mealType,
    required this.calories,
    this.protein,
    this.carbs,
    this.fat,
    required this.mealDate,
    this.notes,
    this.createdAt,
    this.updatedAt,
  });

  factory Meal.fromJson(Map<String, dynamic> json) {
    return Meal(
      id: (json['id'] as num).toInt(),
      name: json['name'] as String,
      mealType: json['mealType'] as String,
      calories: (json['calories'] as num).toInt(),
      protein: json['protein'] != null
          ? (json['protein'] as num).toDouble()
          : null,
      carbs: json['carbs'] != null ? (json['carbs'] as num).toDouble() : null,
      fat: json['fat'] != null ? (json['fat'] as num).toDouble() : null,
      mealDate: DateTime.parse(json['mealDate'] as String),
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'mealType': mealType,
      'calories': calories,
      'protein': protein,
      'carbs': carbs,
      'fat': fat,
      'mealDate': mealDate.toIso8601String(),
      'notes': notes,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}

File: fitness-backend/src/main/java/com/fitness/entity/Meal.java
package com.fitness.entity;

import java.time.LocalDateTime;

import io.quarkus.hibernate.orm.panache.PanacheEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;

@Entity
@Table(name = "meals")
public class Meal extends PanacheEntity {
    
    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    public User user;
    
    @Column(nullable = false)
    public String name; // Yemek adı
    
    @Column(name = "meal_type", nullable = false)
    public String mealType; // BREAKFAST, LUNCH, DINNER, SNACK
    
    @Column(nullable = false)
    public Integer calories; // Kalori
    
    @Column(name = "protein")
    public Double protein; // Protein (gram)
    
    @Column(name = "carbs")
    public Double carbs; // Karbonhidrat (gram)
    
    @Column(name = "fat")
    public Double fat; // Yağ (gram)
    
    @Column(name = "meal_date", nullable = false)
    public LocalDateTime mealDate;
    
    @Column(name = "notes")
    public String notes; // Notlar
    
    @Column(name = "created_at")
    public LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    public LocalDateTime updatedAt;
    
    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (mealDate == null) {
            mealDate = LocalDateTime.now();
        }
    }
    
    @PreUpdate
    public void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}